Tests are failing at TestOIDCReconcile

/kind bug

**What steps did you take and what happened:**

The CI tests (prow job: `pull-cluster-api-provider-aws-test`) in the OIDC tests even without a change in that location:

```
=== RUN   TestOIDCReconcile/cluster_create_with_no_OIDC_provider_present_yet_should_create_one
I0212 18:29:07.593616   46554 oidc.go:56] "Reconciling EKS OIDC Provider" cluster-name="cluster-test"
    iam.go:439: Unexpected call to *mock_iamauth.MockIAMAPI.CreateOpenIDConnectProvider([{
          ClientIDList: ["sts.amazonaws.com"],
          ThumbprintList: ["cfd433f989066ca209571b2a62bb6fa183147112"],
          Url: "https://127.0.0.1:36967"
        }]) at /home/prow/go/src/sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/services/eks/iam/iam.go:439 because: 
        expected call at /home/prow/go/src/sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/services/eks/oidc_test.go:65 doesn't match the argument at index 0.
        Got: {
          ClientIDList: ["sts.amazonaws.com"],
          ThumbprintList: ["cfd433f989066ca209571b2a62bb6fa183147112"],
          Url: "https://127.0.0.1:36967"
        } (*iam.CreateOpenIDConnectProviderInput)
        Want: is equal to {
          ClientIDList: ["sts.amazonaws.com"],
          ThumbprintList: ["15dbd260c7465ecca6de2c0b2181187f66ee0d1a"],
          Url: "https://127.0.0.1:36967"
        } (*iam.CreateOpenIDConnectProviderInput)
    controller.go:269: missing call(s) to *mock_iamauth.MockIAMAPI.CreateOpenIDConnectProvider(is equal to {
          ClientIDList: ["sts.amazonaws.com"],
          ThumbprintList: ["15dbd260c7465ecca6de2c0b2181187f66ee0d1a"],
          Url: "https://127.0.0.1:36967"
        } (*iam.CreateOpenIDConnectProviderInput)) /home/prow/go/src/sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/services/eks/oidc_test.go:65
    controller.go:269: missing call(s) to *mock_iamauth.MockIAMAPI.TagOpenIDConnectProvider(is equal to {
          OpenIDConnectProviderArn: "arn::oidc",
          Tags: []
        } (*iam.TagOpenIDConnectProviderInput)) /home/prow/go/src/sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/services/eks/oidc_test.go:72
    controller.go:269: aborting test due to missing call(s)
=== RUN   TestOIDCReconcile/cluster_create_with_existing_OIDC_provider_which_is_retrieved
I0212 18:29:07.606722   46554 oidc.go:56] "Reconciling EKS OIDC Provider" cluster-name="cluster-test"
    iam.go:439: Unexpected call to *mock_iamauth.MockIAMAPI.CreateOpenIDConnectProvider([{
          ClientIDList: ["sts.amazonaws.com"],
          ThumbprintList: ["cfd433f989066ca209571b2a62bb6fa183147112"],
          Url: "https://127.0.0.1:37871"
        }]) at /home/prow/go/src/sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/services/eks/iam/iam.go:439 because: there are no expected calls of the method "CreateOpenIDConnectProvider" for that receiver
    controller.go:269: missing call(s) to *mock_iamauth.MockIAMAPI.TagOpenIDConnectProvider(is equal to {
          OpenIDConnectProviderArn: "arn::oidc",
          Tags: []
        } (*iam.TagOpenIDConnectProviderInput)) /home/prow/go/src/sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/services/eks/oidc_test.go:108
    controller.go:269: aborting test due to missing call(s)
--- FAIL: TestOIDCReconcile (0.03s)
    --- FAIL: TestOIDCReconcile/cluster_create_with_no_OIDC_provider_present_yet_should_create_one (0.01s)
    --- FAIL: TestOIDCReconcile/cluster_create_with_existing_OIDC_provider_which_is_retrieved (0.01s)
```


**What did you expect to happen:**

I expect the tests to pass

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]


**Environment:**

- Cluster-api-provider-aws version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

--- a/pkg/cloud/services/eks/oidc_test.go
+++ b/pkg/cloud/services/eks/oidc_test.go
@@ -17,6 +17,11 @@ limitations under the License.
 package eks
 
 import (
+	"crypto/sha1"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/hex"
+	"encoding/pem"
 	"net/http"
 	"net/http/httptest"
 	"testing"
@@ -35,10 +40,13 @@ import (
 	ekscontrolplanev1 "sigs.k8s.io/cluster-api-provider-aws/v2/controlplane/eks/api/v1beta2"
 	"sigs.k8s.io/cluster-api-provider-aws/v2/pkg/cloud/scope"
 	"sigs.k8s.io/cluster-api-provider-aws/v2/pkg/cloud/services/iamauth/mock_iamauth"
+	"sigs.k8s.io/cluster-api-provider-aws/v2/pkg/internal/testcert"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 )
 
 func TestOIDCReconcile(t *testing.T) {
+	testCertThumbprint := getTestcertTumbprint(t)
+
 	tests := []struct {
 		name    string
 		expect  func(m *mock_iamauth.MockIAMAPIMockRecorder, url string)
@@ -64,7 +72,7 @@ func TestOIDCReconcile(t *testing.T) {
 				}, nil)
 				m.CreateOpenIDConnectProvider(&iam.CreateOpenIDConnectProviderInput{
 					ClientIDList:   aws.StringSlice([]string{"sts.amazonaws.com"}),
-					ThumbprintList: aws.StringSlice([]string{"15dbd260c7465ecca6de2c0b2181187f66ee0d1a"}),
+					ThumbprintList: aws.StringSlice([]string{testCertThumbprint}),
 					Url:            &url,
 				}).Return(&iam.CreateOpenIDConnectProviderOutput{
 					OpenIDConnectProviderArn: aws.String("arn::oidc"),
@@ -102,7 +110,7 @@ func TestOIDCReconcile(t *testing.T) {
 					OpenIDConnectProviderArn: aws.String("arn::oidc"),
 				}).Return(&iam.GetOpenIDConnectProviderOutput{
 					ClientIDList:   aws.StringSlice([]string{"sts.amazonaws.com"}),
-					ThumbprintList: aws.StringSlice([]string{"15dbd260c7465ecca6de2c0b2181187f66ee0d1a"}),
+					ThumbprintList: aws.StringSlice([]string{testCertThumbprint}),
 					Url:            &url,
 				}, nil)
 				m.TagOpenIDConnectProvider(&iam.TagOpenIDConnectProviderInput{
@@ -125,11 +133,7 @@ func TestOIDCReconcile(t *testing.T) {
 			_ = ekscontrolplanev1.AddToScheme(scheme)
 			_ = corev1.AddToScheme(scheme)
 
-			ts := httptest.NewTLSServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
-				// Send response to be tested
-				rw.WriteHeader(http.StatusOK)
-				rw.Write([]byte(`OK`))
-			}))
+			ts := createTestServer(g)
 			defer ts.Close()
 
 			controlPlane := &ekscontrolplanev1.AWSManagedControlPlane{
@@ -176,6 +180,41 @@ func TestOIDCReconcile(t *testing.T) {
 			g.Expect(err).To(MatchError(ContainSubstring("dial tcp: lookup test-cluster-api.nodomain.example.com")))
 		})
 	}
+}
+
+func getTestcertTumbprint(t *testing.T) string {
+	t.Helper()
+	g := NewWithT(t)
+
+	block, _ := pem.Decode(testcert.LocalhostCert)
+	g.Expect(block).ToNot(BeNil(), "failed to parse certificate PEM")
+
+	x509Cert, err := x509.ParseCertificate(block.Bytes)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	hash := sha1.Sum(x509Cert.Raw) //nolint:gosec
+	return hex.EncodeToString(hash[:])
+}
+
+func createTestServer(g *GomegaWithT) *httptest.Server {
+	// Create a certificate and private key
+	cert, err := tls.X509KeyPair(testcert.LocalhostCert, testcert.LocalhostKey)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	// Create custom TLS config
+	tlsConfig := &tls.Config{ //nolint:gosec
+		Certificates: []tls.Certificate{cert},
+	}
+
+	// Create test server with custom TLS config
+	server := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte(`OK`))
+	}))
+	server.TLS = tlsConfig
+	server.StartTLS()
+
+	return server
 }
 
 var kubeConfig = []byte(`apiVersion: v1

--- a/pkg/internal/testcert/testcert.go
+++ b/pkg/internal/testcert/testcert.go
@@ -1,0 +1,62 @@
+// Package testcert contains a test-only localhost certificate.
+// NOTE: this is based on the code from the Go stdlib.
+package testcert
+
+import "strings"
+
+// LocalhostCert is a PEM-encoded TLS cert with SAN IPs
+// "127.0.0.1" and "[::1]", expiring at Jan 29 16:00:00 2084 GMT.
+// generated from src/crypto/tls:
+// go run generate_cert.go  --rsa-bits 2048 --host 127.0.0.1,::1,example.com --ca --start-date "Jan 1 00:00:00 1970" --duration=1000000h.
+var LocalhostCert = []byte(`-----BEGIN CERTIFICATE-----
+MIIDOTCCAiGgAwIBAgIQSRJrEpBGFc7tNb1fb5pKFzANBgkqhkiG9w0BAQsFADAS
+MRAwDgYDVQQKEwdBY21lIENvMCAXDTcwMDEwMTAwMDAwMFoYDzIwODQwMTI5MTYw
+MDAwWjASMRAwDgYDVQQKEwdBY21lIENvMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8A
+MIIBCgKCAQEA6Gba5tHV1dAKouAaXO3/ebDUU4rvwCUg/CNaJ2PT5xLD4N1Vcb8r
+bFSW2HXKq+MPfVdwIKR/1DczEoAGf/JWQTW7EgzlXrCd3rlajEX2D73faWJekD0U
+aUgz5vtrTXZ90BQL7WvRICd7FlEZ6FPOcPlumiyNmzUqtwGhO+9ad1W5BqJaRI6P
+YfouNkwR6Na4TzSj5BrqUfP0FwDizKSJ0XXmh8g8G9mtwxOSN3Ru1QFc61Xyeluk
+POGKBV/q6RBNklTNe0gI8usUMlYyoC7ytppNMW7X2vodAelSu25jgx2anj9fDVZu
+h7AXF5+4nJS4AAt0n1lNY7nGSsdZas8PbQIDAQABo4GIMIGFMA4GA1UdDwEB/wQE
+AwICpDATBgNVHSUEDDAKBggrBgEFBQcDATAPBgNVHRMBAf8EBTADAQH/MB0GA1Ud
+DgQWBBStsdjh3/JCXXYlQryOrL4Sh7BW5TAuBgNVHREEJzAlggtleGFtcGxlLmNv
+bYcEfwAAAYcQAAAAAAAAAAAAAAAAAAAAATANBgkqhkiG9w0BAQsFAAOCAQEAxWGI
+5NhpF3nwwy/4yB4i/CwwSpLrWUa70NyhvprUBC50PxiXav1TeDzwzLx/o5HyNwsv
+cxv3HdkLW59i/0SlJSrNnWdfZ19oTcS+6PtLoVyISgtyN6DpkKpdG1cOkW3Cy2P2
++tK/tKHRP1Y/Ra0RiDpOAmqn0gCOFGz8+lqDIor/T7MTpibL3IxqWfPrvfVRHL3B
+grw/ZQTTIVjjh4JBSW3WyWgNo/ikC1lrVxzl4iPUGptxT36Cr7Zk2Bsg0XqwbOvK
+5d+NTDREkSnUbie4GeutujmX3Dsx88UiV6UY/4lHJa6I5leHUNOHahRbpbWeOfs/
+WkBKOclmOV2xlTVuPw==
+-----END CERTIFICATE-----`)
+
+// LocalhostKey is the private key for LocalhostCert.
+var LocalhostKey = []byte(testingKey(`-----BEGIN RSA TESTING KEY-----
+MIIEvAIBADANBgkqhkiG9w0BAQEFAASCBKYwggSiAgEAAoIBAQDoZtrm0dXV0Aqi
+4Bpc7f95sNRTiu/AJSD8I1onY9PnEsPg3VVxvytsVJbYdcqr4w99V3AgpH/UNzMS
+gAZ/8lZBNbsSDOVesJ3euVqMRfYPvd9pYl6QPRRpSDPm+2tNdn3QFAvta9EgJ3sW
+URnoU85w+W6aLI2bNSq3AaE771p3VbkGolpEjo9h+i42TBHo1rhPNKPkGupR8/QX
+AOLMpInRdeaHyDwb2a3DE5I3dG7VAVzrVfJ6W6Q84YoFX+rpEE2SVM17SAjy6xQy
+VjKgLvK2mk0xbtfa+h0B6VK7bmODHZqeP18NVm6HsBcXn7iclLgAC3SfWU1jucZK
+x1lqzw9tAgMBAAECggEABWzxS1Y2wckblnXY57Z+sl6YdmLV+gxj2r8Qib7g4ZIk
+lIlWR1OJNfw7kU4eryib4fc6nOh6O4AWZyYqAK6tqNQSS/eVG0LQTLTTEldHyVJL
+dvBe+MsUQOj4nTndZW+QvFzbcm2D8lY5n2nBSxU5ypVoKZ1EqQzytFcLZpTN7d89
+EPj0qDyrV4NZlWAwL1AygCwnlwhMQjXEalVF1ylXwU3QzyZ/6MgvF6d3SSUlh+sq
+XefuyigXw484cQQgbzopv6niMOmGP3of+yV4JQqUSb3IDmmT68XjGd2Dkxl4iPki
+6ZwXf3CCi+c+i/zVEcufgZ3SLf8D99kUGE7v7fZ6AQKBgQD1ZX3RAla9hIhxCf+O
+3D+I1j2LMrdjAh0ZKKqwMR4JnHX3mjQI6LwqIctPWTU8wYFECSh9klEclSdCa64s
+uI/GNpcqPXejd0cAAdqHEEeG5sHMDt0oFSurL4lyud0GtZvwlzLuwEweuDtvT9cJ
+Wfvl86uyO36IW8JdvUprYDctrQKBgQDycZ697qutBieZlGkHpnYWUAeImVA878sJ
+w44NuXHvMxBPz+lbJGAg8Cn8fcxNAPqHIraK+kx3po8cZGQywKHUWsxi23ozHoxo
++bGqeQb9U661TnfdDspIXia+xilZt3mm5BPzOUuRqlh4Y9SOBpSWRmEhyw76w4ZP
+OPxjWYAgwQKBgA/FehSYxeJgRjSdo+MWnK66tjHgDJE8bYpUZsP0JC4R9DL5oiaA
+brd2fI6Y+SbyeNBallObt8LSgzdtnEAbjIH8uDJqyOmknNePRvAvR6mP4xyuR+Bv
+m+Lgp0DMWTw5J9CKpydZDItc49T/mJ5tPhdFVd+am0NAQnmr1MCZ6nHxAoGABS3Y
+LkaC9FdFUUqSU8+Chkd/YbOkuyiENdkvl6t2e52jo5DVc1T7mLiIrRQi4SI8N9bN
+/3oJWCT+uaSLX2ouCtNFunblzWHBrhxnZzTeqVq4SLc8aESAnbslKL4i8/+vYZlN
+s8xtiNcSvL+lMsOBORSXzpj/4Ot8WwTkn1qyGgECgYBKNTypzAHeLE6yVadFp3nQ
+Ckq9yzvP/ib05rvgbvrne00YeOxqJ9gtTrzgh7koqJyX1L4NwdkEza4ilDWpucn0
+xiUZS4SoaJq6ZvcBYS62Yr1t8n09iG47YL8ibgtmH3L+svaotvpVxVK+d7BLevA/
+ZboOWVe3icTy64BT3OQhmg==
+-----END RSA TESTING KEY-----`))
+
+func testingKey(s string) string { return strings.ReplaceAll(s, "TESTING KEY", "PRIVATE KEY") }

